pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderBase.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.authentication;
+
+import io.opentelemetry.api.OpenTelemetry;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.pulsar.broker.authentication.metrics.AuthenticationMetrics;
+
+public abstract class AuthenticationProviderBase implements AuthenticationProvider {
+
+    @Getter
+    private AuthenticationMetrics authenticationMetrics;
+
+    protected final void initializeMetrics(@NonNull OpenTelemetry openTelemetry) {
+        authenticationMetrics = new AuthenticationMetrics(openTelemetry);
+    }
```

Comment:
Doing a quick scan of this PR so my review comments aren't well thought out. :)
Instead of making this class the preferred base class for AuthenticationProviders, I think that it would be better to pass an initialized AuthenticationMetrics instance directly to the AuthenticationProvider interface.
I think that a context pattern would be better so that there's an initialization method in AuthenticationProvider that gets passed a context which can be used to lookup various "services" that the AuthenticationProvider needs.
the context could have a getter for the AuthenticationMetrics and OpenTelemetry (if it has custom metrics).
the `void initialize(ServiceConfiguration config) throws IOException;` method in AuthenticationProvider should be deprecated and replaced with the `void initialize(AuthenticationProviderContext context);` method.
`AuthenticationProviderContext` would have the methods to get ServiceConfiguration, AuthenticationMetrics, OpenTelemetry and new methods could be later on added when needed.

Reply:
Added `AuthenticationProvider.Context` class to hold relevant variables for initializing the AuthenticationProvider. Included the `ServiceConfiguration` and `OpenTelemetry` objects. Opted against initializing the `AuthenticationMetrics` object outside of the `Context`, since different authentication methods can have different metrics and this decision fits better with the provider; it can also complicate testing.

Removed class `AuthenticationProviderBase`, as it was providing little benefit.

Let me know if this looks better.